qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -813,6 +814,10 @@ def _definitely_skip_node(
             dag._has_calibration_for(node)
             or len(node.qargs) < self._min_qubits
             or node.is_directive()
+            or (
+                self._instruction_supported(node.name, qubits)
+                and node.name not in CONTROL_FLOW_OP_NAMES
```

Comment:
How about using `node.is_control_flow()`?

Reply:
Thanks! Done in [47cbfdf](https://github.com/Qiskit/qiskit/pull/13417/commits/47cbfdf25a094f3de4e8bdbd58c4c450fb44ec40)